### ISSUE TITLE
[Q] general queries related to markdown usage in Tizen documentation

### ISSUE THREAD

**Issue by journey2w** [Samsung/tizen-docs#50]
Issued by @l10nbridge.

----------------------------------------------------

The following issues are general queries related to markdown usage in Tizen documentation:

1.	“<” and “>” characters:
When exactly do these need to be replaced with `&lt;` and `&gt;`? 
We are now assuming that the actual “<” and “>” characters are only allowed inside inline code and code snippets, and need to be changed everywhere else. Is that right?

2.	Notes (and tips and warnings and similar callouts)
We have implemented these as follows:
```
> **Note** (2 spaces are added to the end of the line to force a line break)
> Actual note text
````
Is this OK? The instructions in the GitHub site use “> [!Note]” formatting, but that does not show up correctly in the browser (and has not been used in any of the converted topics)…

3.	Links leading to  a subsection within a topic
We have implemented these as follows:
```
“[Link text](topic.md#anchor)”
```
In the subsection heading: `Heading text<a name=”anchor”></a>`

Is that OK?
Also, when linking to  a subheading within the same topic, is it necessary to add the topic file name before “#anchor”?

4.	Trademark symbols
We have been currently ignoring this issue in all the files we have edited for GitHub.
We need to know how the trademark symbols must be implemented. Is `&trade;` and `&reg;` OK? Or do we need to use the actual symbol characters ™ and ®, even though they look quite corrupted in the browser…

5.	Backslash \ and vertical bar |
Are these characters supposed to be used as “\\” and “\ |” everywhere, except in code snippets of inline code?
They seem to work without the extra backslash in notes too…

6.	Code blocks
What are the allowed (and required) language codes that need to be added to the beginning of a code block?
```
  ```<language>
  code text
  ```

The GitHub site mentions “xml”, “json”, “csharp”, “bash” and “output”.
What needs to be used for JavaScript, EDC, and HTML? Or for generic config file content?

And if the code snippet shows both a CLI command and the command results, is the correct language code then “bash” or “output”, or something else?

7.	Numbered lists
We are creating numbered list by using just number “1”, to ease later maintenance.
We hope this is OK.

Do we need to do anything special with step lists embedded inside other step lists to make the embedded list use “a, b, c” numbering instead of “1,2,3”? At the moment, all numbered lists, embedded or not, show 1,2,3. 

8.	Superscript
How is superscript text created?

9.	Highlight in code blocks
How do we create a highlight for specific parts of text within a code block? (This was created with a highlight class and span element in HTML…)

10.	Outside links
Previously, all links to outside third party Web sites were created with the target=”_blank” attribute, which ensured that the clicked link opened in a new window in a browser. Is there functionality like that in Markdown?

11.	Image width limitations
What are the image width limitations on the new site? Are we still supposed to check and modify images that are wider than 850 pixels?
Also, if the TOC box has been removed, does it mean that we do not need to worry about images at the top of the topic anymore, and the separate 450 pixel limitation no longer exists?
We have not touched oversized images in GitHub yet.

12.	Tables
Is any kind of table cell merging allowed anymore? If yes, how should it be implemented?

If a table cell requires text with line breaks, we are implementing those with `<br>` . Is that OK?

If a table cell requires lists, we are using `<ul>…</ul>` or `<ol>…</ol>`. Is that OK?

If a table cell requires inline code or a code block, how is that implemented?

13.	Confusing instructions in the GitHub site:
- The instructions say that titles should use sentence case (only first letter of first word capitalized).  Our current library uses title case (all first letters capitalized). Do we need to change all the titles in all our topics?

- The instructions say that images should use red rounded-rectangle outlines for important issues, and 1-pixel gray edges for white bleeding areas. The images in our current files use non-rounded red rectangles, and 1- or 2-pixel (it varies) black edges. Does this mean that we need to go through and update all our images? Or is this a rule to be followed in the future? Or does this rule only apply to other content than our files?

- The instructions say that image alt text needs a meaningful description along with SEO terms. Currently, we simply copy the image caption text to the alt text field. Do all the existing alt texts need to be modified? Or is this rule something we need to follow in the future somehow?

- The instructions say “In general, if you don't see something described here, look in editing markdown files for examples.”
What does this mean? Is there an “editing markdown files” guide somewhere, or does this text try to say that the existing markdown files in the master/live branches should be used as examples?

**Comment by l10nbridge** [comment]
I am updating this list based on answers and new queries generated from PR #52:

1. Based on the info, we are going to replace  “<” and “>” characters with `&lt;` and `&gt;` everywhere, except inside inline code and code snippets.

2. Apparently, notes are supposed to be created with an empty line between the `**Note**` and the actual note content. We have been using 2 spaces after `**Note**` and no empty line, but we will fix those in the topics we already edited.

3. Links leading to a subsection within a topic.  
This issue seems a bit complicated.
   - First of all, if we have a new topic that has not been rendered yet, how do we confirm the header ID?
   - Secondly, if the links are implemented with IDs that are not included in the MD files, how are we going to run any kind of link checks? Who exactly is going to be responsible for checking that all links within the library work?

   We could have checked the links if they had been implemented as we first thought, with the `<a name=”anchor”></a>` tags added after the headings. But if that is not allowed, we do not know how to verify that links actually work.

   Please clarify how exactly we can verify that all links work in this system?

   By the way, we are now assuming that it is NOT necessary to add the topic file name before “#ID” in the link. And we will be removing those from the topics we already edited. 

4. Trademark symbols
We assume that using `&trade;` and `&reg;`is OK, and will fix all topics we have already edited accordingly.

5. Backslash \ and vertical bar |
This issue is closed.

6. Code blocks

   - Some of the content we have been editing has been using "c++" as a language. We will change those to  "cpp". We will also remove all "output" instances used as language.
   -  We still need clarification on what is the correct language text to be used for HTML, C, and EDC code examples.
   - If the code snippet shows both a CLI command and the command results, is the correct language code then “bash” or nothing?
   - Some of the files we have received for editing have used "sh" as a language for a code block. Is that OK?

7.  Numbered lists
This issue is closed.

8.    Superscript
How is superscript text created?

9.    Highlight in code blocks
    How do we create a highlight for specific parts of text within a code block? (This was created with a highlight class and span element in HTML…)

10.    Outside links
This issue is closed.

11.    Image width limitations
    What are the image width limitations on the new site? Are we still supposed to check and modify images that are wider than 850 pixels?
    Also, if the TOC box has been removed, does it mean that we do not need to worry about images at the top of the topic anymore, and the separate 450 pixel limitation no longer exists?
    We have not touched oversized images in GitHub yet.

12. Tables

     -  Is any kind of table cell merging allowed anymore? If yes, how should it be implemented?
     - If a table cell requires text with line breaks, we are implementing those with `<br>` . Is that OK?
     - If a table cell requires lists, we are using `<ul>…</ul>` or `<ol>…</ol>`. Is that OK?
     - If a table cell requires inline code or a code block, how is that implemented?

13. Confusing instructions in the GitHub site:
Remaining issues:
     - The instructions say that titles should use sentence case (only first letter of first word capitalized). Our current library uses title case (all first letters capitalized). Do we need to change all the titles in all our topics?
     - The instructions say that images should use red rounded-rectangle outlines for important issues, and 1-pixel gray edges for white bleeding areas. The images in our current files use non-rounded red rectangles, and 1- or 2-pixel (it varies) black edges. Does this mean that we need to go through and update all our images? Or is this a rule to be followed in the future? Or does this rule only apply to other content than our files?

14. New issues based on #52:

    - What are the rules for using the "horizontal rule" `---`?
       We have not used this kind of dividing line before, and cannot think of a reason why to use it in the future. Please clarify exactly in what kind of circumstances it is needed. Otherwise, we will not use it, and will delete it in all the topics we edit.
    - The new writing rules state that italics is supposed to be used for files/folders/paths/new terms.
      In the current documentation, we have avoided italics completely, since it tends to be hard to read from an electronic screen. Currently we are using inline code to write files/folders/paths, and use either quotes and bold font for new terms, as needed.
      Are we now really supposed to follow this new rule and change every single file/folder/path/new term instance to use italics?
    - The new writing rules are confusing, They state that an automatic list of second level heading links will be generated as an "In this article" section at the beginning. However, the template guide still contains a separate "Content" section. 
      So what exactly is the plan? Are we supposed to provide some kind of "content" section separately (and add it manually) in all the topics, while the system also generates an automatic "In this article" section? This makes no sense.
    (By the way, we did not know about the automatically generated "In this article" issue, so some of the topics we have already edited now contain a link list at the beginning linking to the subsections (a bit like a "content" section, though not named that way. So based on what the plan here actually is, some changes will be needed in the topics we already edited.)

**Comment by journey2w** [comment]
Fixed #52
